Stripe/STPPaymentContext.swift

```diff
@@ -0,0 +1,1129 @@
+      }
+    }
+  }
+
+  @objc
+  public func paymentOptionsViewController(
+    _ paymentOptionsViewController: STPPaymentOptionsViewController,
+    didFailToLoadWithError error: Error
+  ) {
+    // we'll handle this ourselves when the loading promise fails.
+  }
+
+  @objc(appropriatelyDismissPaymentOptionsViewController:completion:) func appropriatelyDismiss(
+    _ viewController: STPPaymentOptionsViewController,
+    completion: @escaping STPVoidBlock
+  ) {
+    if viewController.stp_isAtRootOfNavigationController() {
+      // if we're the root of the navigation controller, we've been presented modally.
+      viewController.presentingViewController?.dismiss(
+        animated: transitionAnimationsEnabled()
+      ) {
+        self.paymentOptionsViewController = nil
+        completion()
+      }
+    } else {
+      // otherwise, we've been pushed onto the stack.
+      var destinationViewController = hostViewController
+      // If hostViewController is a nav controller, pop to the original VC on top of the stack.
+      if hostViewController is UINavigationController {
+        destinationViewController = originalTopViewController
+      }
+      viewController.navigationController?.stp_pop(
+        to: destinationViewController,
+        animated: transitionAnimationsEnabled()
+      ) {
+        self.paymentOptionsViewController = nil
+        completion()
+      }
+    }
+  }
+
+  // MARK: - Shipping Info
+
+  func presentShippingViewController(withNewState state: STPPaymentContextState) {
+    assert(
+      hostViewController != nil && hostViewController?.viewIfLoaded?.window != nil,
+      "hostViewController must not be nil on STPPaymentContext when calling presentShippingViewController on it. Next time, set the hostViewController property first!"
+    )
+
+    if self.state == STPPaymentContextState.none {
+      self.state = state
+
+      let addressViewController = STPShippingAddressViewController(paymentContext: self)
+      addressViewController.navigationItem.largeTitleDisplayMode = largeTitleDisplayMode
+      let navigationController = UINavigationController(rootViewController: addressViewController)
+      navigationController.navigationBar.stp_theme = theme
+      navigationController.navigationBar.prefersLargeTitles = true
+      navigationController.modalPresentationStyle = modalPresentationStyle
+      hostViewController?.present(
+        navigationController,
+        animated: transitionAnimationsEnabled())
+    }
+  }
+
+  @objc
+  public func shippingAddressViewControllerDidCancel(
+    _ addressViewController: STPShippingAddressViewController
+  ) {
+    appropriatelyDismiss(addressViewController) {
+      if self.state == .requestingPayment {
+        self.didFinish(
+          with: .userCancellation,
+          error: nil)
+      } else {
+        self.state = STPPaymentContextState.none
+      }
+    }
+  }
+
+  @objc
+  public func shippingAddressViewController(
```

Comment:
Should not be public

Reply:
Implementations of public protocols need to be public